### DISCUSSION TITLE
Submitting jobs on a cluster with MaxArraySize

### DISCUSSION THREAD

**Discussion by YaraToenders** [HALFpipe/HALFpipe#438] [General] [answered]
**What happened?**
I am trying to submit a job with 188 subjects to a cluster where the MaxArraySize is 120. Following this explanation: https://stackoverflow.com/questions/69543503/start-a-sbatch-array-with-big-number, I added the following lines to slurm.submit.sh:
#SBATCH --array=0-68

VALUES=({121..188})
THISJOBVALUE=${VALUES[$SLURM_TASK_ARRAY_ID]}

singularity run \
--contain --cleanenv \
--bind /:/ext \
/path/to/halfpipe-halfpipe-1.2.2.sif \
--workdir /path/to/workdir/ \
--only-run \
--uuid 4f6619b6 \
--subject-chunks \
--only-chunk-index ${THISJOBVALUE} \
--nipype-n-procs 2 \
--keep some

This should avoid going over the max array and still running these subjects. The submission works, however, it seems to only run subject 121 (from the log file:
grep -hnr "Will run subject level chunk" log.txt 
826:[2022-09-26 10:44:08,0202] [halfpipe        ] [INFO     ] Will run subject level chunk 121 of 188
840:[2022-09-26 10:44:08,0418] [halfpipe        ] [INFO     ] Will run subject level chunk 121 of 188
848:[2022-09-26 10:44:08,0572] [halfpipe        ] [INFO     ] Will run subject level chunk 121 of 188
849:[2022-09-26 10:44:08,0611] [halfpipe        ] [INFO     ] Will run subject level chunk 121 of 188
850:[2022-09-26 10:44:08,0630] [halfpipe        ] [INFO     ] Will run subject level chunk 121 of 188
851:[2022-09-26 10:44:08,0641] [halfpipe        ] [INFO     ] Will run subject level chunk 121 of 188
1169:[2022-09-26 10:44:08,0662] [halfpipe        ] [INFO     ] Will run subject level chunk 121 of 188
1170:[2022-09-26 10:44:08,0678] [halfpipe        ] [INFO     ] Will run subject level chunk 121 of 188
1171:[2022-09-26 10:44:08,0714] [halfpipe        ] [INFO     ] Will run subject level chunk 121 of 188
1172:[2022-09-26 10:44:08,0718] [halfpipe        ] [INFO     ] Will run subject level chunk 121 of 188
1173:[2022-09-26 10:44:08,0773] [halfpipe        ] [INFO     ] Will run subject level chunk 121 of 188
2450:[2022-09-26 10:44:08,0717] [halfpipe        ] [INFO     ] Will run subject level chunk 121 of 188
2774:[2022-09-26 10:44:08,0602] [halfpipe        ] [INFO     ] Will run subject level chunk 121 of 188
3092:[2022-09-26 10:44:08,0577] [halfpipe        ] [INFO     ] Will run subject level chunk 121 of 188
)

Do you have any alternative solutions to avoiding the MaxArraySize problem that might work?

**What is the exact command that you used to run `Halfpipe`?**
sbatch submit.slurm.sh

**Which settings did you use?**
Standard ENIGMA settings
[log.txt](https://github.com/HALFpipe/HALFpipe/files/9644965/log.txt)

**Comment by HippocampusGirl** [comment]
Hi Yara,

Have you tried running with just 
```bash
 #SBATCH --array=121-188
```

I can have a more detailed look once I'm back from vacation :-)

**Comment by YaraToenders** [comment]
Hi Lea,
Thanks! That's what I tried at first but it gives an error when submitting it. The cluster is down today, but I'll post the error tomorrow!

**Comment by YaraToenders** [comment]
Hi Lea,

The problem is that when using` #SBATCH --array=121-188` I am not able to submit the job:
```
sbatch submit.slurm.sh 
sbatch: error: Batch job submission failed: Invalid job array specification
```
I had hoped that adding the values would have solved this, but for some reason it doesn't seem to work and I can't figure out why. Thanks again for your time! (and enjoy your holiday!)

**Comment by YaraToenders** [comment]
Hi Lea, 

Have you had a chance to have a look at this?
Thanks!

**Reply by HippocampusGirl** [reply]
Hi Yara,

It's actually really surprising for me, since this works on our cluster here in Berlin. Would you be available for a short video call to show me what's going on?

**Comment by HippocampusGirl** [comment, marked as answer]
We figured out the issue! It was a typo, it needs to be `SLURM_ARRAY_TASK_ID` not `SLURM_TASK_ARRAY_ID`

**Comment by YaraToenders** [comment]
Thanks for your help!
